### ISSUE TITLE
totally incorrect advice on permissions

### ISSUE THREAD

**Issue by f1-outsourcing** [piwik/piwik#20583]
```
ote: Matomo's One Click update requires write-permissions to the Matomo folder and its contents.

Matomo is unable to overwrite your current installation. You can either fix the directory/file permissions, or download the package and install version 4.14.1 manually:
chown -R xxxxx:xxxxxx/home/xxxxxx/stats.xxxxxxx.xxx
find /home/xxxxxxx/stats.xxxxxx.xxxx -type f -exec chmod 644 {} \;
find /home/xxxxxxxx/stats.xxxxxx.xxx -type d -exec chmod 755 {} \;
chmod 755 /home/xxxxxxxxx/stats.xxxxxx.com/console
After you execute these commands (or change permissions via your FTP software), refresh the page and you should be able to use the "Automatic Update" feature. 
```

You can't just give chmod 644 out of the blue, in my case others is always 0. First maybe check what the current permissions of files/dirs are that are ok, and then maybe advise those to use. Now you are advising something insecure.

**Comment by f1-outsourcing** [comment]
Or maybe do like wordpress ftps update so you are sure the website files cannot be changed.

**Comment by mattab** [comment]
Here is our official secure recommendation currently found [within this guide](https://matomo.org/faq/on-premise/how-to-configure-matomo-for-security/):

* **Make as few files and directories writable by your webserver as possible.** 

For Matomo to work best, we recommend giving your webserver write access to these files and directories within your Matomo root directory: `config/config.ini.php`, the `misc/user` directory, `matomo.js`, `piwik.js` and `tmp` directory. Only the `tmp` directory is absolutely required. A few features might not work if you have only the `tmp` directory writable. Once you have limited file permissions, you won't be able to use the one-click updater through the UI anymore. Also it won't be possible to install any plugin from the Marketplace through the UI. We therefore recommend you configure the setting `enable_auto_update = 0` within the `[General]` section of your `config/config.ini.php` file. You can then instead [update Matomo manually](https://matomo.org/docs/update/#the-manual-three-step-update) and [install plugins manually](https://matomo.org/faq/plugins/faq_21/).


:arrow_up: What do you think about this documentation above? 

For Matomo to give better information to users, maybe we could 
1. create a  FAQ that has just this help text in it
2. link to this FAQ from the product when the error occurs

**Comment by f1-outsourcing** [comment]
Yes I would indeed refrain from advising with a specific umask. Then again, nobody reads, so it is nice to have some copy past example. 
What also can be done is 
Is trying to test before giving the advice thus, 
one can assume that files are read correctly otherwise the web interface would not be working. 
1. read the default set umask, the webhoster most likely has prepared this correctly for the website.
2. read the one of the current files, and add +w to the group if there is none, otherwise add +w to others if there is none.
  if this would be not enough in the group, then a reload of the test page would show the 2nd time the advise command to change others to +w

Side note is of course when acls are used. I can't imagine there is not a nice little script that does this all.
